services/device-registry/src/test/java/org/eclipse/hono/deviceregistry/FileBasedRegistrationServiceTest.java

```diff
@@ -45,405 +48,417 @@
-    * the registration service to write its content to the file system during shutdown.
-    * 
-    * @param ctx The vert.x test context.
-    */
-   @SuppressWarnings({ "unchecked", "rawtypes" })
-   @Test
-   public void testContentsNotSavedOnShutdownIfSavingIfDisabled(final TestContext ctx) {
-
-       // GIVEN a registration service configured to not persist data
-       props.setSaveToFile(false);
-       when(fileSystem.existsBlocking(props.getFilename())).thenReturn(Boolean.TRUE);
-       doAnswer(invocation -> {
-           Handler handler = invocation.getArgumentAt(1, Handler.class);
-           handler.handle(Future.failedFuture("malformed data"));
-           return null;
-       }).when(fileSystem).readFile(eq(props.getFilename()), any(Handler.class));
-       Async startup = ctx.async();
-       Future<Void> startupTracker = Future.future();
-       startupTracker.setHandler(ctx.asyncAssertSuccess(started -> {
-           startup.complete();
-       }));
-       registrationService.doStart(startupTracker);
-       startup.await(2000);
-
-       // WHEN adding a device
-       registrationService.addDevice(TENANT, DEVICE, new JsonObject());
-       // and shutting down the service
-       Async shutdown = ctx.async();
-       Future<Void> shutdownTracker = Future.future();
-       shutdownTracker.setHandler(ctx.asyncAssertSuccess(done -> {
-           shutdown.complete();
-       }));
-       registrationService.doStop(shutdownTracker);
-
-       // THEN no data has been written to the file system
-       shutdown.await(2000);
-       verify(fileSystem, never()).createFile(eq(props.getFilename()), any(Handler.class));
-   }
-
-   private static JsonObject expectedPayload(final String id) {
-       return new JsonObject()
-               .put(FIELD_DEVICE_ID, id)
-               .put(FIELD_DATA, new JsonObject().put(FIELD_ENABLED, Boolean.TRUE));
-   }
-
-   private static Message<JsonObject> mockMsg(final String action) {
-       return mockMsg(action, TENANT);
-   }
-
-   @SuppressWarnings("unchecked")
-   private static Message<JsonObject> mockMsg(final String action, final String tenant) {
-       final JsonObject registrationJson = getServiceRequestAsJson(action, tenant, DEVICE);
-       final Message<JsonObject> message = mock(Message.class);
-       when(message.body()).thenReturn(registrationJson);
-       return message;
-   }
-
-   private void processMessageAndExpectResponse(final Message<JsonObject> request, final JsonObject expectedResponse) {
-       registrationService.processRegistrationMessage(request);
-       verify(request).reply(expectedResponse);
-   }
+public class FileBasedRegistrationServiceTest {
+
+    private static final String TENANT = Constants.DEFAULT_TENANT;
+    private static final String DEVICE = "4711";
+    private static final String GW = "gw-1";
+    private static final String FILE_NAME = "/device-identities.json";
+
+    /**
+     * Time out all tests after 5 secs.
+     */
+    @Rule
```

Comment:
@ClassRule was used in other test class - why @Rule here?

Reply:
My understanding is that `@Classrule` defines the maximum execution time for **all** test cases together instead of fore **each** test case. But I may be mistaken.

Reply:
Yes, you are right, my understanding is the same.
Maybe the comment for the `@Classrule` should be extended to something like
`Global timeout for all test cases in summary.`

Reply:
will replace `ClassRule` with `Rule`